seal_fhe/src/encryptor_decryptor.rs

```diff
@@ -115,6 +119,46 @@ impl Encryptor {
 
         Ok(ciphertext)
     }
+
+    /**
+     *
+     * Encrypts a plaintext with the public key and returns the ciphertext as a
+     * serializable object. Also returns the u and e values used in encrypting
+     * the value.
+     *
+     * The encryption parameters for the resulting ciphertext correspond to:
+     * 1) in BFV, the highest (data) level in the modulus switching chain,
+     * 2) in CKKS, the encryption parameters of the plaintext.
+     * Dynamic memory allocations in the process are allocated from the memory
+     * pool pointed to by the given MemoryPoolHandle.
+     *
+     * * `plainext` - The plaintext to encrypt.
+     * * `disable_special_modulus` - Whether to disable the special modulus switching
+     */
+    pub fn encrypt_return_components(
```

Comment:
`encrypt_return_noise` is probably a better name

Reply:
Well r is not noise; it is the component that handles that seal rounds the message instead of just flooring it.

Reply:
Hmm... the thing we want to indicate is "encrypt and return the data needed to make a ZKP." Maybe "encrypt_and_return_zkp_data"? component is a very generic term.